src/coreclr/jit/compiler.h

```diff
@@ -11945,7 +11945,7 @@ class GenTreeVisitor
 
                 if (call->gtCallType == CT_INDIRECT)
                 {
-                    if (call->gtCallCookie != nullptr)
+                    if ((call->gtCallCookie != nullptr) && !call->IsVirtualStub())
```

Comment:
If I understand correctly, the active member of the union is not `gtCallCookie` when `call->IsVirtualStub()`, but rather either `gtInlineCandidateInfo` or `gtHandleHistogramProfileCandidateInfo`. If so that makes it undefined behavior to access `gtCallCookie` in these cases.

I think all the `call->IsVirtualStub()` checks need to be first to avoid this UB.